docs/plugin/messaging.md

```diff
@@ -0,0 +1,886 @@
+In the past, prior to the R3 version of Thunder, logging, tracing and warning reporting were completely separate issues and were handled individually. However, even though these message types have distinct characteristics, we recognized the advantages of consolidating them into a unified framework, which is now referred to as `Messaging`. In the early versions of R3, we still had tracing enabled by default and messaging was in the early development. It is the R4 version of Thunder where `Messaging` is finally the default framework to handle all types of messages.
+
+
+# Advantages of using messaging
+
+There are of course several advantages of using this framework to deliver the messages over simply using something like `printf`:
+
+* There are not only different types of messages, but also categories within these types, and each of them can be enabled/disabled at runtime
+* No serialization penalty between different processes
+* Always getting full lines one after another since all messages are timestamped
+* Possibility of redirecting the messages to another output like, e.g., a console, syslog, a file or a network stream, which is particularly useful in the case of containers, and this is something we are working towards
+* In the near future, *standard out* and *standard error* will also be redirectable
+
+We are convinced that `Messaging` is much better suited for the development, and with some recent changes which will be described in the following sections, it is fully operational and enabled by default. We strongly believe that it is a good time to shed some more light on that functionality, since it is simple to use and yet very effective.
+
+
+# Differences between the message types
+
+Logging, tracing and warning reporting are important techniques often used in software development to gather information and provide insight into the behavior of an application. However, there are some fundamental differences between them within Thunder, in a nutshell:
+
+* Tracing is meant for the developers and is dropped in production
```

Comment:
I think we should be explicit and say "dropped in production by default" as if I understand correctly it is still possible to enable Tracing in production if required?

Also what determines a production build?

Reply:
In `Production`, so when building with the `Min_Size_Rel` flag, the `TRACE` and `TRACE_GLOBAL` macros are declared empty and there is no way to enable tracing. This version of Thunder is meant to be used only by operators who want the smallest footprint on memory possible.

In both `Debug` and `Release` versions, tracing is enabled. `Debug` is used by developers, so all macros are enabled and it has no code optimization at all, whereas `Release` is used by most of our operators and the QA team - asserts off and some code optimization.